src/CfgMgr32/CfgMgr32.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright © .NET Foundation and Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Exported functions from the NTDll.dll Windows library
+    /// that are available to Desktop apps only.
+    /// </content>
+    [OfferFriendlyOverloads]
+    public static partial class CfgMgr32
+    {
+        /// <summary>
+        /// The callback routine invoked by the Configuration Manager when listening for events.
+        /// </summary>
+        /// <param name="notify">
+        /// The handle of the notification context which invoked the callback.
+        /// </param>
+        /// <param name="context">
+        /// A user-provided callback context.
+        /// </param>
+        /// <param name="action">
+        /// The device action which triggered the callback.
+        /// </param>
+        /// <param name="eventData">
+        /// The callback data.
+        /// </param>
+        /// <param name="eventDataSize">
+        /// The size of the callback data struct.
+        /// </param>
+        /// <returns>
+        /// If responding to a <see cref="CM_NOTIFY_ACTION.DEVICEQUERYREMOVE"/> notification,
+        /// the <see cref="CM_NOTIFY_CALLBACK"/> callback should return either <c>ERROR_SUCCESS</c> or <c>ERROR_CANCELLED</c>,
+        /// as appropriate. Otherwise, the callback should return <c>ERROR_SUCCESS</c>.
+        /// The callback should not return any other values.
+        /// </returns>
+        public delegate uint CM_NOTIFY_CALLBACK(
+            IntPtr notify,
+            IntPtr context,
```

Comment:
I can't find the docs for `CM_NOTIFY_CALLBACK` but are these pointers, values or black boxes? If pointers please use C# pointers. Same for `eventData`.

Reply:
The documentation is here: https://docs.microsoft.com/en-us/windows/win32/api/cfgmgr32/nf-cfgmgr32-cm_register_notification#remarks . `eventData` is of type [CM_NOTIFY_EVENT_DATA](https://docs.microsoft.com/en-us/windows/win32/api/cfgmgr32/ns-cfgmgr32-cm_notify_event_data), which is a `union`. How do you handle unions in this project?

Reply:
We handle unions with `[StructLayout(Explicit)]` on the struct.